### DISCUSSION TITLE
Should Kubernetes have semantics for traffic profiles?

### DISCUSSION THREAD

**Discussion by arnatal** [CloudNativeSDWAN/cnwan-operator#55] [General]
As part of the CN-WAN presentation in KubeCon NA, we raised the question: "Should Kubernetes have semantics in general for traffic profiles?". 

Our motivation for asking this question comes from our experience with CN-WAN. We have found really helpful to have annotations that specify the *traffic profile* for an application (i.e. the type of traffic that the application generates or consumes). In the CN-WAN case, those *traffic profiles* help the CN-WAN components to optimize the application traffic over the WAN connection. We wonder if the model could be generalized and apply to Kubernetes applications in general, so that application traffic can be optimized not only on the WAN but in any network (e.g. cloud, DC, campus, etc) based on the *traffic profile*.

We would love to hear opinions on this idea, so we are opening this GitHub issue to collect the thoughts from the community.

**Comment by asimpleidea** [comment]
Pinning this.

From a technical perspective, a *CRD* named *TrafficProfile* would be introduced and the CN-WAN Operator would implement a controller that reconciles them.

**Comment by arnatal** [comment]
> From a technical perspective, a _CRD_ named _TrafficProfile_ would be introduced and the CN-WAN Operator would implement a controller that reconciles them.

Yes, that's one way we could look at this. I'm curious on how the community sees this and which other approaches we should consider.

**Comment by asimpleidea** [comment]
Converting this into a discussion as I think it wasn't really fitting as an issue.
